src/main/java/Duke.java

```diff
@@ -1,10 +1,91 @@
+import duke.DukeException;
+import duke.command.Command;
+import duke.main.Parser;
+import duke.main.Storage;
+import duke.main.TaskList;
+import duke.main.Ui;
+import java.io.IOException;
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private static final Scanner scan = new Scanner( System.in );
+    private static Storage storage;
+    private static TaskList tasks;
+    private static Ui ui;
+
+    /**
+     * Default constructor to fix javafx issue.
+     */
+    public Duke() { }
+
+    /**
+     * Constructor to initialise Duke using the save file.
+     *
+     * @param filePath Path where the save file is stored.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        try {
+            storage = new Storage(filePath);
+            tasks = new TaskList(storage.loadInTasks());
+        } catch (IOException ioe) {
+            System.exit(0);
+        } catch (DukeException de) {
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Main method to initialise Duke.
+     *
+     * @param args The user input.
+     * @throws DukeException If invalid command encountered.
+     */
+    public static void main(String[] args) throws DukeException {
+        new Duke("data/tasks.txt").run();
```

Comment:
Since the data file will be a fixed location, perhaps can consider extracting it out as a static variable to improve readability?